src/components/modal/modal.js

```diff
@@ -600,11 +604,15 @@ export const BModal = /*#__PURE__*/ Vue.extend({
     onClose() {
       this.hide('headerclose')
     },
-    onEsc(evt) {
+    onKeyDown(evt) {
       // If ESC pressed, hide modal
       if (evt.keyCode === KeyCodes.ESC && this.isVisible && !this.noCloseOnEsc) {
         this.hide('esc')
       }
+      // If Enter pressed, submit modal
+      if (evt.keyCode === KeyCodes.ENTER && this.isVisible && !this.noSubmitOnEnter) {
+        this.onOk()
+      }
```

Comment:
This would close the modal even in there are cancel buttons, etc.

Reply:
Would it ok to pass the default action via props?

Reply:
It would probably be better to have some option(s) to specify which default rendered buttons have focus when the modal opens (defaulting to no buttons having focus, as autofocusing an input or button on a modal will usually only announce the button/input but not the modal content when using screen readers.

A more robust approach is to have users listen for the `shown` event, and then focus the element they want (be it a button, input, or other focusable content control).

The `shown` event includes a reference to the modal element in the bvEvent object (the `target` property).  Once shown, you can then use `bvEvent.target.querySelector('some selector criteria`)` to find an element and focus it.